tests/glm_ostream_operators.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * tests/glm_ostream_operators.h
+ *
+ * Copyright 2018 Endless Mobile, Inc.
+ *
+ * libanimation is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 2.1 of the
+ * License, or (at your option) any later version.
+ *
+ * libanimation is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with eos-companion-app-service.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ *
+ * A simple helper to output the contents of glm types when
+ * tests fail.
+ */
+#pragma once
+
+#include <glm/glm.hpp>
+#include <iomanip>                      // for operator<<, setprecision
+#include <iosfwd>                       // for ostream
+#include <ostream>                      // for basic_ostream, char_traits, etc
+
+namespace
+{
+    namespace detail
+    {
+        template <typename T, int D>
+        struct ArrayDimensionPrinter
+        {
+            static std::ostream & Apply (std::ostream &lhs,
+                                         T      const &array)
```

Comment:
Weird alignment (also below)

Reply:
Ah this is actually intentional (and should be consistent with the rest of the code). I'm using the "type modifiers bind right" style here, eg `Foo const &name`